crypto/keys/generate.go

```diff
@@ -34,38 +28,6 @@ const (
 	MinRsaKeySizeInBits = 2048
 )
 
-// NewFromSpec generates a new private key based on a key specification.
-// If an RSA key is specified, the key size must be at least MinRsaKeySizeInBits.
-func NewFromSpec(spec *keyspb.Specification) (crypto.Signer, error) {
-	switch params := spec.GetParams().(type) {
-	case *keyspb.Specification_EcdsaParams:
-		curve := ECDSACurveFromParams(params.EcdsaParams)
-		if curve == nil {
-			return nil, fmt.Errorf("unsupported ECDSA curve: %s", params.EcdsaParams.GetCurve())
-		}
-
-		return ecdsa.GenerateKey(curve, rand.Reader)
-
-	case *keyspb.Specification_RsaParams:
-		bits := int(params.RsaParams.GetBits())
-		if bits == 0 {
-			bits = DefaultRsaKeySizeInBits
-		}
-		if bits < MinRsaKeySizeInBits {
-			return nil, fmt.Errorf("minimum RSA key size is %v bits, got %v bits", MinRsaKeySizeInBits, bits)
-		}
-
-		return rsa.GenerateKey(rand.Reader, bits)
-
-	case *keyspb.Specification_Ed25519Params:
-		_, privKey, err := ed25519.GenerateKey(rand.Reader)
-		return privKey, err
-
-	default:
-		return nil, fmt.Errorf("unsupported keygen params type: %T", params)
-	}
-}
-
 // ECDSACurveFromParams returns the curve specified by the given parameters.
```

Comment:
Is this used outside of `NewFromSpec`? Remove?

Reply:
Same for the consts above, and this file as a whole.

Reply:
It's still used elsewhere, and I'll move them out to more appropriate locations in a follow up PR.